tsdb/mapper.go

```diff
@@ -436,10 +485,10 @@ type tagSetCursor struct {
 	cursors     []*seriesCursor   // Underlying series cursors.
 	decoder     *FieldCodec       // decoder for the raw data bytes
 
-	// Lookahead buffers for the cursors. Performance analysis shows that it is critical
-	// that these buffers are part of the tagSetCursor type and not part of the cursors type.
-	keyBuffer   []int64  // The current timestamp key for each cursor
-	valueBuffer [][]byte // The current value for each cursor
+	// Lookahead queue of points from the cursors, implemented using a heap.
+	// Performance profiling shows that it is critical that this queue is part
+	// of the tagSetCursor type and not part of the the cursors type.
+	pqueue *rawPoints
```

Comment:
Should this be named "heap"? Queue makes me think that pushed items come off in the same order as they were put on.